internal/consensus/state.go

```diff
@@ -2338,7 +2395,7 @@ func (cs *State) addProposalBlockPart(msg *BlockPartMessage, peerID p2p.ID) (add
 			cs.ProposalBlockParts.ByteSize(), maxBytes,
 		)
 	}
-	if added && cs.ProposalBlockParts.IsComplete() {
+	if added && cs.ProposalBlockParts.IsComplete() && cs.ProposalBlobParts.IsComplete() {
```

Comment:
Will we ever get here again if we do not have all the blob parts but we do all the block parts? 

That is, say you have all the block parts, you will never get here again, so you will never buil the block?

I would propose to move the check `cs.ProposalBlobParts.IsComplete()` to line 2419 and if the proposal is really complete fire the event, otherwise we get out of this functon and fire it when we have all the blob parts. (Would still have to check this makes sense once looking at other parts of the code).

Reply:
Good point, new commit does that.